core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -337,6 +338,8 @@ object KafkaConfig {
   val SslClientAuthProp = SslConfigs.SSL_CLIENT_AUTH_CONFIG
 
   /** ********* SASL Configuration ****************/
+  val SaslMechanismProp = SaslConfigs.SASL_MECHANISM
```

Comment:
I asked @junrao offline about this and he suggested the `sasl.mechanism.inter.broker.protocol`. The reason why I think this is different than `sasl.kerberos.service.name` is that it's easy to be confused when one sees `sasl.mechanism` and `sasl.enabled.mechanism`.